src/coord/src/catalog.rs

```diff
@@ -349,22 +354,27 @@ impl Catalog {
             nonce: rand::random(),
             experimental_mode,
             cluster_id,
+            /// https://www.postgresql.org/message-id/2108.972178856@sss.pgh.pa.us
+            oid_counter: 16384,
```

Comment:
Can you extract this into a counter like `FIRST_USER_OID` and start it at 20_000? We already use 16384 for the list type; that’s hard coded somewhere in the pgrepr package. (Long term we need a better assignment strategy for system type OIDs, but just making some space should do the trick for now.)